### PR TITLE
Support alignment in sceKernelAllocMemBlock

### PR DESCRIPTION
**Pull request by sunho**
In the saekano game, the following error was preventing to load the opening video before entering  the loading scene after the logo scene.

Screen Shot 2020-06-05 at 1 15 33 PM

It turned out that the video decoder allocate the memory block by non-null sceKernelAllocMemBlockOpt with alignment option. The game crashes manually when it found out the block address is not aligned.
 
Provided sceKernelAllocMemBlockOpt:
Screen Shot 2020-06-05 at 1 23 15 PM
 
This PR add the memory alignment implementation. With this PR, the game can now finish the  logo scene and enter the loading scene.

### CODE REVIEW

File: vita3k/mem/src/mem.cpp [hunk 1]
```diff
@@ -116,6 +116,17 @@ bool init(MemState &state) {
...
 }
 
+Address alloc(MemState &state, size_t size, const char *name, unsigned int alignment) {
+    if (alignment != 0)
+        size += alignment - 1;
+    auto addr = alloc(state, size, name);
+    if (alignment == 0)
+        return addr;
+    auto new_addr = (addr + alignment - 1) & ~(alignment - 1);
+    state.aligned_addr_to_original[addr] = new_addr;
```
Comment (frangarcj) [inline vita3k/mem/src/mem.cpp:126]: Shouldn't this be the opposite?

Addr_to_original[new_addr]=addr ?

File: vita3k/renderer/src/gl/texture.cpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ static size_t decompress_compressed_swizz_texture(SceGxmTextureBaseFormat fmt, v
         return (((width + 3) / 4) * ((height + 3) / 4) * ((ubc_type > 1) ? 16 : 8));
     } else if ((fmt >= SCE_GXM_TEXTURE_BASE_FORMAT_PVRT2BPP) && (fmt <= SCE_GXM_TEXTURE_BASE_FORMAT_PVRTII4BPP)) {
         // TODO, is not perfect for PVRT-II.
-        pvr::PVRTDecompressPVRTC(data, (fmt == SCE_GXM_TEXTURE_BASE_FORMAT_PVRT2BPP) || (fmt == SCE_GXM_TEXTURE_BASE_FORMAT_PVRTII2BPP), width, height, 
+        pvr::PVRTDecompressPVRTC(data, (fmt == SCE_GXM_TEXTURE_BASE_FORMAT_PVRT2BPP) || (fmt == SCE_GXM_TEXTURE_BASE_FORMAT_PVRTII2BPP), width, height,
```
Comment (frangarcj) [inline vita3k/renderer/src/gl/texture.cpp:112]: This doesn't belong to this pr

File: vita3k/mem/src/mem.cpp (hunk 1 above)
Comment (sunho) [reply vita3k/mem/src/mem.cpp:126]: Oh yes. :O

File: vita3k/mem/src/mem.cpp [hunk 3]
```diff
@@ -142,6 +153,12 @@ Address alloc_at(MemState &state, Address address, size_t size, const char *name
 }
 
 void free(MemState &state, Address address) {
+    auto &addr_map = state.aligned_addr_to_original;
+    if (addr_map.find(address) != addr_map.end()) {
```
Comment (frangarcj) [inline vita3k/mem/src/mem.cpp:157]: Is it mandatory to do the find or you can just access with [old_addr]?

Comment (sunho) [reply vita3k/mem/src/mem.cpp:157]: It's mandatory since I didn't set the non aligned addr to the map.
